### PR TITLE
2022 CDN Chapter

### LINKED ISSUE #2900: CDN 2022
# CDN 2022

CDN illustration

If you're interested in contributing to the CDN chapter of the 2022 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| ---- | ------- | --------- | -------- | ------- | ----------- |
| @harendra | @harendra @joeviggiano | @ytkoka | @harendra @joeviggiano | @tunetheweb | @paulcalvano |

<details>
<summary><b>Expand for more information about each role 👀</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 1**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **May 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 1**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **June 1 - 15**: HTTP Archive runs the June crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **August 1**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [x] **September 1**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [x] **September 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **September 26**: Target launch date 🚀

## Chapter resources

Refer to these 2022 CDN resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata
[💬 #web-almanac-cdn on Slack][~slack-channel] for team coordination

[~google-doc]: https://docs.google.com/document/d/1lTrBZGE00XsiFesalFJg5zCzhtMbsrJ5CcdH2eQ_HKE/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2022/cdn
[~google-sheets]: https://docs.google.com/spreadsheets/d/1ySETT2IZ9ae5_VUphxUol2ZU3P1RJvcSVjDU5BgnK5A/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2022/cdn.md
[~slack-channel]: https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA

### PR DESCRIPTION
**Pull request by harendra**
Closes #2900 

CDN Markdown created. Following changes were made.

1. New cdn.md created for year 2022
2. 2022.json changed to add CDN chapter and authors.
3. Image files added in static folder.

Staged URL: https://20221012t112755-dot-webalmanac.uk.r.appspot.com/en/2022/cdn
Featured-Quote: https://20221012t112755-dot-webalmanac.uk.r.appspot.com/en/2022/?feat=cdn#featured-chapter

### CODE REVIEW

**Comment by tunetheweb** [comment]
@harendra, @joeviggiano, @ytkoka I've copy edited the chapter and made it available here:

Staged URL: https://20221012t112755-dot-webalmanac.uk.r.appspot.com/en/2022/cdn
Featured-Quote: https://20221012t112755-dot-webalmanac.uk.r.appspot.com/en/2022/?feat=cdn#featured-chapter

The main changes I made, that you can see in https://github.com/HTTPArchive/almanac.httparchive.org/pull/3212/commits/64cfde6732ef7dadd697f2cd87660d60350ef662, were:
- Rewrote the top CDNs query to make it easier to query across years
- Removed yourselves as editors and added myself (authors are expected to edit their own work, but should have an independent person actually do the final copy edit)
- Remove the desktop data. Appreciate that was done last year, but it read as repetitive.
- Removed references to the increase in HTTP Archive domains - that actually came after the Web Almanac June dataset we used, so one for next year!
- Move commentary to after the figures, rather than before - as per the Web Almanac style
- Cleaned up some of the figures
- Other misc edits for readability

It would be great if you could review ASAP so we can include this chapter in the next follow up launch.

Thanks again for stepping up to take on this chapter at short notice!

File: src/content/en/2022/cdn.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+* This increase was not seen in 2021 as many businesses were still trying to figure out the optimal solution for their problem. In fact we saw a dip in CDN usage for the subdomain and third-party type.
+* Sites relied on serving third-party content through third-party domains instead of their own domains. The fact that the amount of content served from third-party domains increased by 3% during this period supports this assumption.
+
+Regarding the base HTML page, the traditional pattern has been to serve the base HTML from the origin and this pattern has continued as majority of base pages continue to be served from the origin. However, there has been a 4% increase in the base pages being served from CDNs. The trend of base HTML pages being served from the CDN is clearly on the rise.
+
+These are some of the likely reasons behind the rise:
+* CDNs can improve load time of the base HTML page that can be of high importance to improve customer experience and keep users engaged.
+* It is simpler and faster to load the base page from CDNs using the DNS provided by CDN providers. {# TODO - is it? Why? #}
```
Comment (tunetheweb) [inline src/content/en/2022/cdn.md:128]: @harendra @joeviggiano I've a query on this statement.

Comment (harendra) [reply src/content/en/2022/cdn.md:128]: @tunetheweb Thanks for reviewing this. Regarding your query, let me put the reason, Distributed nature of CDNs make recursive query response faster.
(https://www.cloudflare.com/learning/dns/what-is-1.1.1.1/).  For eg. Amazon R53 does not require you to CNAME to CloudFront reducing extra DNS hops, this also adds to the performance. If you want to serve the site via apex domain you generally need to host it on DNS provided by CDN, for the users it's simpler to just host their domains in CDN providers.

Comment (harendra) [reply src/content/en/2022/cdn.md:128]: @tunetheweb Thanks, yea that works.

**Pull request merged**